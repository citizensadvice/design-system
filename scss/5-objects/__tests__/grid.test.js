/* eslint-env jest */
const sass = require('sass');

test('grid styles match snapshot', () => {
  const output = sass.renderSync({
    file: 'scss/5-objects/_grid.scss',
  });
  expect(output.css.toString()).toMatchSnapshot();
});
