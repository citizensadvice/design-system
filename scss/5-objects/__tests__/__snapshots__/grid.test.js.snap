// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`grid styles match snapshot 1`] = `
"/* @use must be first, which conflicts with the import position rule */
/* stylelint-disable no-invalid-position-at-import-rule */
.cads-grid-container {
  width: 100%;
  margin-right: auto;
  margin-left: auto;
  padding-right: 1rem;
  padding-left: 1rem;
}
@media (min-width: 768px) {
  .cads-grid-container {
    padding-right: 2rem;
    padding-left: 2rem;
  }
}
@media (min-width: 1024px) {
  .cads-grid-container {
    padding-right: 2.5rem;
    padding-left: 2.5rem;
  }
}
@media (min-width: 1024px) {
  .cads-grid-container {
    max-width: 1090px;
  }
}

.cads-grid-row {
  display: flex;
  flex-wrap: wrap;
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}
@media (min-width: 768px) {
  .cads-grid-row {
    margin-right: -1rem;
    margin-left: -1rem;
  }
}
@media (min-width: 1024px) {
  .cads-grid-row {
    margin-right: -1.25rem;
    margin-left: -1.25rem;
  }
}

.cads-grid-col-lg,
.cads-grid-col-lg-auto, .cads-grid-col-lg-12, .cads-grid-col-lg-11, .cads-grid-col-lg-10, .cads-grid-col-lg-9, .cads-grid-col-lg-8, .cads-grid-col-lg-7, .cads-grid-col-lg-6, .cads-grid-col-lg-5, .cads-grid-col-lg-4, .cads-grid-col-lg-3, .cads-grid-col-lg-2, .cads-grid-col-lg-1, .cads-grid-col-md,
.cads-grid-col-md-auto, .cads-grid-col-md-12, .cads-grid-col-md-11, .cads-grid-col-md-10, .cads-grid-col-md-9, .cads-grid-col-md-8, .cads-grid-col-md-7, .cads-grid-col-md-6, .cads-grid-col-md-5, .cads-grid-col-md-4, .cads-grid-col-md-3, .cads-grid-col-md-2, .cads-grid-col-md-1, .cads-grid-col,
.cads-grid-col-auto, .cads-grid-col-12, .cads-grid-col-11, .cads-grid-col-10, .cads-grid-col-9, .cads-grid-col-8, .cads-grid-col-7, .cads-grid-col-6, .cads-grid-col-5, .cads-grid-col-4, .cads-grid-col-3, .cads-grid-col-2, .cads-grid-col-1 {
  position: relative;
  width: 100%;
  min-height: 1px;
}

.cads-grid-col-lg,
.cads-grid-col-lg-auto, .cads-grid-col-lg-12, .cads-grid-col-lg-11, .cads-grid-col-lg-10, .cads-grid-col-lg-9, .cads-grid-col-lg-8, .cads-grid-col-lg-7, .cads-grid-col-lg-6, .cads-grid-col-lg-5, .cads-grid-col-lg-4, .cads-grid-col-lg-3, .cads-grid-col-lg-2, .cads-grid-col-lg-1, .cads-grid-col-md,
.cads-grid-col-md-auto, .cads-grid-col-md-12, .cads-grid-col-md-11, .cads-grid-col-md-10, .cads-grid-col-md-9, .cads-grid-col-md-8, .cads-grid-col-md-7, .cads-grid-col-md-6, .cads-grid-col-md-5, .cads-grid-col-md-4, .cads-grid-col-md-3, .cads-grid-col-md-2, .cads-grid-col-md-1, .cads-grid-col,
.cads-grid-col-auto, .cads-grid-col-12, .cads-grid-col-11, .cads-grid-col-10, .cads-grid-col-9, .cads-grid-col-8, .cads-grid-col-7, .cads-grid-col-6, .cads-grid-col-5, .cads-grid-col-4, .cads-grid-col-3, .cads-grid-col-2, .cads-grid-col-1 {
  padding-right: 0.5rem;
  padding-left: 0.5rem;
}

.cads-grid-col {
  flex-basis: 0;
  flex-grow: 1;
  max-width: 100%;
}

.cads-grid-col-auto {
  flex: 0 0 auto;
  width: auto;
  max-width: none;
}

.cads-grid-col-1 {
  flex: 0 0 8.3333333333%;
  max-width: 8.3333333333%;
}

.cads-grid-col-2 {
  flex: 0 0 16.6666666667%;
  max-width: 16.6666666667%;
}

.cads-grid-col-3 {
  flex: 0 0 25%;
  max-width: 25%;
}

.cads-grid-col-4 {
  flex: 0 0 33.3333333333%;
  max-width: 33.3333333333%;
}

.cads-grid-col-5 {
  flex: 0 0 41.6666666667%;
  max-width: 41.6666666667%;
}

.cads-grid-col-6 {
  flex: 0 0 50%;
  max-width: 50%;
}

.cads-grid-col-7 {
  flex: 0 0 58.3333333333%;
  max-width: 58.3333333333%;
}

.cads-grid-col-8 {
  flex: 0 0 66.6666666667%;
  max-width: 66.6666666667%;
}

.cads-grid-col-9 {
  flex: 0 0 75%;
  max-width: 75%;
}

.cads-grid-col-10 {
  flex: 0 0 83.3333333333%;
  max-width: 83.3333333333%;
}

.cads-grid-col-11 {
  flex: 0 0 91.6666666667%;
  max-width: 91.6666666667%;
}

.cads-grid-col-12 {
  flex: 0 0 100%;
  max-width: 100%;
}

@media (min-width: 768px) {
  .cads-grid-col-lg,
.cads-grid-col-lg-auto, .cads-grid-col-lg-12, .cads-grid-col-lg-11, .cads-grid-col-lg-10, .cads-grid-col-lg-9, .cads-grid-col-lg-8, .cads-grid-col-lg-7, .cads-grid-col-lg-6, .cads-grid-col-lg-5, .cads-grid-col-lg-4, .cads-grid-col-lg-3, .cads-grid-col-lg-2, .cads-grid-col-lg-1, .cads-grid-col-md,
.cads-grid-col-md-auto, .cads-grid-col-md-12, .cads-grid-col-md-11, .cads-grid-col-md-10, .cads-grid-col-md-9, .cads-grid-col-md-8, .cads-grid-col-md-7, .cads-grid-col-md-6, .cads-grid-col-md-5, .cads-grid-col-md-4, .cads-grid-col-md-3, .cads-grid-col-md-2, .cads-grid-col-md-1, .cads-grid-col-1, .cads-grid-col-2, .cads-grid-col-3, .cads-grid-col-4, .cads-grid-col-5, .cads-grid-col-6, .cads-grid-col-7, .cads-grid-col-8, .cads-grid-col-9, .cads-grid-col-10, .cads-grid-col-11, .cads-grid-col-12, .cads-grid-col,
.cads-grid-col-auto {
    padding-right: 1rem;
    padding-left: 1rem;
  }
}

@media (min-width: 768px) {
  .cads-grid-col-md {
    flex-basis: 0;
    flex-grow: 1;
    max-width: 100%;
  }
  .cads-grid-col-md-auto {
    flex: 0 0 auto;
    width: auto;
    max-width: none;
  }
  .cads-grid-col-md-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .cads-grid-col-md-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .cads-grid-col-md-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .cads-grid-col-md-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .cads-grid-col-md-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .cads-grid-col-md-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .cads-grid-col-md-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .cads-grid-col-md-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .cads-grid-col-md-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .cads-grid-col-md-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .cads-grid-col-md-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .cads-grid-col-md-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 1024px) {
  .cads-grid-col-lg,
.cads-grid-col-lg-auto, .cads-grid-col-lg-12, .cads-grid-col-lg-11, .cads-grid-col-lg-10, .cads-grid-col-lg-9, .cads-grid-col-lg-8, .cads-grid-col-lg-7, .cads-grid-col-lg-6, .cads-grid-col-lg-5, .cads-grid-col-lg-4, .cads-grid-col-lg-3, .cads-grid-col-lg-2, .cads-grid-col-lg-1, .cads-grid-col-1, .cads-grid-col-2, .cads-grid-col-3, .cads-grid-col-4, .cads-grid-col-5, .cads-grid-col-6, .cads-grid-col-7, .cads-grid-col-8, .cads-grid-col-9, .cads-grid-col-10, .cads-grid-col-11, .cads-grid-col-12, .cads-grid-col,
.cads-grid-col-auto, .cads-grid-col-md-1, .cads-grid-col-md-2, .cads-grid-col-md-3, .cads-grid-col-md-4, .cads-grid-col-md-5, .cads-grid-col-md-6, .cads-grid-col-md-7, .cads-grid-col-md-8, .cads-grid-col-md-9, .cads-grid-col-md-10, .cads-grid-col-md-11, .cads-grid-col-md-12, .cads-grid-col-md,
.cads-grid-col-md-auto {
    padding-right: 1.25rem;
    padding-left: 1.25rem;
  }
}

@media (min-width: 1024px) {
  .cads-grid-col-lg {
    flex-basis: 0;
    flex-grow: 1;
    max-width: 100%;
  }
  .cads-grid-col-lg-auto {
    flex: 0 0 auto;
    width: auto;
    max-width: none;
  }
  .cads-grid-col-lg-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .cads-grid-col-lg-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .cads-grid-col-lg-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .cads-grid-col-lg-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .cads-grid-col-lg-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .cads-grid-col-lg-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .cads-grid-col-lg-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .cads-grid-col-lg-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .cads-grid-col-lg-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .cads-grid-col-lg-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .cads-grid-col-lg-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .cads-grid-col-lg-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}"
`;
